### PR TITLE
docs: update the headers usage in gen command.

### PR DESCRIPTION
**Pull request by laststylebender14**

### CODE REVIEW

File: docs/cli.md [hunk 1]
```diff
@@ -133,13 +138,24 @@ To generate a TailCall GraphQL configuration, provide a configuration file to th
...
 
    ```json
    {
      "curl": {
        "src": "https://jsonplaceholder.typicode.com/posts/1",
-       "fieldName": "post"
+       "fieldName": "post",
+       "headers": {
+         "Content-Type": "application/json",
+         "Accept": "application/json",
```
Comment (tusharmath) [inline docs/cli.md]:
```suggestion
```

File: docs/cli.md [hunk 2]
```diff
@@ -133,13 +138,24 @@ To generate a TailCall GraphQL configuration, provide a configuration file to th
 
 The `inputs` section specifies the sources from which the GraphQL configuration should be generated. Each source can be either a REST endpoint or a protobuf file.
 
-1. **REST:** For REST endpoints, provide the endpoint URL (`src`) and the field name in the configuration (`fieldName`), which will be used as the field name in the operation type.
+1. **REST:** When defining REST endpoints, the configuration should include the endpoint URL (src), the required headers for making the HTTP request, and the field name (fieldName) that will be used as the field name in the operation type.
+
+   1. **Headers:** Users can specify the required headers to make the HTTP request in the headers section.
```
Comment (tusharmath) [inline docs/cli.md]: Between different fields — src, fieldName, headers - We should specify what's required and what's optional and what are the default values.

File: docs/cli.md [hunk 3]
```diff
@@ -106,7 +106,12 @@ To generate a TailCall GraphQL configuration, provide a configuration file to th
     {
       "curl": {
         "src": "https://jsonplaceholder.typicode.com/posts/1",
-        "fieldName": "post"
+        "fieldName": "post",
+        "headers": {
+          "Content-Type": "application/json",
+          "Accept": "application/json",
+          "Authorization": "Bearer {{.env.AUTH_TOKEN}}"
+        }
```
Comment (tusharmath) [inline docs/cli.md:119]: If YAML works, can you add a tab for JSON version and its YAML Version here.

File: docs/cli.md (hunk 1 above)
Comment (laststylebender14) [reply docs/cli.md]: done.

File: docs/cli.md (hunk 3 above)
Comment (laststylebender14) [reply docs/cli.md:119]: done.

File: docs/cli.md (hunk 2 above)
Comment (laststylebender14) [reply docs/cli.md]: done.

**Review by tusharmath — APPROVED** [review]

**Pull request merged**